### PR TITLE
complete the controller context for init funcs

### PR DESCRIPTION
**Pull request by deads2k**
This completes the conversion to initFuncs for the controller initialization to make easier and more manageable to add them.

### CODE REVIEW

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -274,11 +281,6 @@ func KnownControllers() []string {
 
 	ret.Insert(
```
Comment (liggitt) [inline cmd/kube-controller-manager/app/controllermanager.go]: move the comment about special controllers here, and give guidance if people should not generally add anything else to this list

Comment (deads2k) [inline cmd/kube-controller-manager/app/controllermanager.go]: done

**Comment by falenn** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  curtis.l.bates@gmail.com
    Domain imwiz.com has exceeded the max emails per hour (180/150 (120%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
------ The body of the message is 10440 characters long; only the first
------ 5000 or so are included here.
Received: from github-smtp2-ext2.iad.github.net ([192.30.252.193]:55530 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by box969.bluehost.com with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.87)
	(envelope-from <noreply@github.com>)
	id 1dGTPL-000Wvr-8g
	for dev@imwiz.com; Thu, 01 Jun 2017 10:57:11 -0600
Date: Thu, 01 Jun 2017 09:57:00 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1496336220;
	bh=E1/LKYroBkAIeMRViOT+4U1wkfnqgj8U9t7cGARbyiY=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=Hdb7qZpxldCgxRmlJ5i+3DlE9T3te5cmmnTgDKxLkCDhNJoK6jYkYtQuWH4VXt9y0
	 U+ptBqps5LTs/tXZIiH8bqktgPFZzDIFvfVCNCIkgzmjkUndCs79UTWiRGdBoszi+z
	 TeppIhCM8GxcqQpFA5hqGxWmxzCkrXdJVz2DFTrg=

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 2]
```diff
@@ -452,144 +436,57 @@ func StartControllers(controllers map[string]InitFunc, s *options.CMServer, root
...
+		serviceaccountcontroller.TokensControllerOptions{
+			TokenGenerator: serviceaccount.JWTTokenGenerator(privateKey),
+			RootCA:         rootCA,
+		},
+	)
+	go controller.Run(int(ctx.Options.ConcurrentSATokenSyncs), ctx.Stop)
 
-	sharedInformers.Start(stop)
+	// start the first set of informers now so that other controllers can start
+	ctx.InformerFactory.Start(ctx.Stop)
```
Comment (liggitt) [inline cmd/kube-controller-manager/app/controllermanager.go]: is it odd to tie sharedInformers to this controller? if you're not running with `--use-service-account-credentials`, what else in the controller manager depends on this controller?

Comment (deads2k) [inline cmd/kube-controller-manager/app/controllermanager.go]: > is it odd to tie sharedInformers to this controller?

Not really, this is the logical order from before, it's just slightly more obvious

Comment (deads2k) [inline cmd/kube-controller-manager/app/controllermanager.go]: > what else in the controller manager depends on this controller?

Not completely certain.  I think the future is subdivision.

**Comment by deads2k** [comment]
@k8s-bot pull-kubernetes-unit test this

**Comment by deads2k** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this
@k8s-bot pull-kubernetes-kubemark-e2e-gce test this

**Comment by deads2k** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by deads2k** [comment]
@k8s-bot pull-kubernetes-unit test this

**Comment by deads2k** [comment]
@k8s-bot pull-kubernetes-kubemark-e2e-gce test this

**Comment by deads2k** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this
@k8s-bot pull-kubernetes-kubemark-e2e-gce test this
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by deads2k** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by deads2k** [comment]
Queue is confused.  Etcd3 passed

**Comment by liggitt** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by deads2k** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by fejta** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this
ref: #46827

**Pull request merged**